### PR TITLE
feat: decouple cosmos-db

### PR DESCRIPTION
**Pull request by cool-develope**
## Context

As discussed in [this](https://github.com/cosmos/cosmos-sdk/issues/19005), we need to decouple `cosmos-db` from `iavl`.

- Implemented the own raw DB interface
- Added the `Wrapper` to support the old clients which use `cosmos-db`
- cleanup
- updated golangci-lint from 1.51.2 to 1.55.2  [ref](https://github.com/golang/go/issues/64592)

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]
nit: I would put all DB types into public `const`.

File: db/memdb.go [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+func newKey(key []byte) item {
+	return item{key: key}
+}
+
+// newPair creates a new pair item.
+func newPair(key, value []byte) item {
+	return item{key: key, value: value}
+}
+
+// MemDB is an in-memory database backend using a B-tree for storage.
```
Comment (tac0turtle) [inline db/memdb.go:40]: would mention this is meant for testing

File: nodedb.go [hunk 2]
```diff
@@ -327,13 +328,13 @@ func (ndb *nodeDB) saveFastNodeUnlocked(node *fastnode.Node, shouldAddToCache bo
...
-		}
-		return exists, nil
-	}
+	// if ldb, ok := ndb.db.(*dbm.GoLevelDB); ok {
+	// 	exists, err := ldb.DB().Has(key, nil)
+	// 	if err != nil {
+	// 		return false, err
+	// 	}
+	// 	return exists, nil
+	// }
```
Comment (tac0turtle) [inline nodedb.go:337]: remove?

**Review by tac0turtle — APPROVED** [review]

**Comment by cool-develope** [comment]
@tac0turtle backport to v1.0.x?

**Comment by alexanderbez** [comment]
@mergifyio backport release/v1.x.x

**Pull request merged**